packages/core/src/external/aws/s3.ts

```diff
@@ -152,6 +158,47 @@ export class S3Utils {
     }
   }
 
+  async fileExists(bucket: string, key: string): Promise<boolean>;
+  async fileExists(bucket: string, filters: FileExistsFilter): Promise<boolean>;
+  async fileExists(bucket: string, keyOrFilters: string | FileExistsFilter): Promise<boolean> {
+    if (typeof keyOrFilters === "string") {
+      const fileInfo = await this.getFileInfoFromS3(keyOrFilters, bucket);
+      return fileInfo.exists;
+    }
+    return this.filesWithPathExist({
+      bucket,
+      ...keyOrFilters,
+    });
+  }
+
+  private async filesWithPathExist({
+    bucket,
+    path,
+    targetString,
+  }: {
+    bucket: string;
+    path: string;
+    targetString?: string | undefined;
+  }): Promise<boolean> {
+    const data = await executeWithRetriesS3(() =>
+      this._s3Client.send(
+        new ListObjectsV2Command({
+          Bucket: bucket,
+          Prefix: path,
+        })
+      )
+    );
+    const contents = data.Contents;
+    if (!contents) return false;
+
+    for (const file of contents) {
+      if (targetString && file.Key?.includes(targetString)) {
+        return true;
+      }
+    }
```

Comment:
Why are we checking the contents of the file? Don't we just need to check the name?

Reply:
yea, `file.Key` is the name

Reply:
ok so it was a little confusing when reading this. Could we change the variable names:
contents -> bucketContents
file -> fileName

Reply:
bucketContents ok, but file is NOT fileName.. `file.Key` = fileName